clients/launcher/lib/widgets/welcome_modal.dart

```diff
@@ -110,191 +188,631 @@ class _WelcomeModalContentState extends State<_WelcomeModalContent> {
                 children: [
-                  SailButton.secondary(
-                    'Generate Random',
-                    onPressed: _handleGenerateRandom,
-                    size: ButtonSize.regular,
+                  // Input field for entropy/hex
+                  Row(
+                    children: [
+                      Expanded(
+                        child: SizedBox(
+                          height: 24,
+                          child: TextField(
+                            controller: _entropyController,
+                            style: const TextStyle(fontSize: 10, color: Colors.white),
+                            decoration: const InputDecoration(
+                              isDense: true,
+                              contentPadding: EdgeInsets.symmetric(horizontal: 4, vertical: 4),
+                              hintText: 'Enter any text or hex',
+                              hintStyle: TextStyle(fontSize: 10, color: Colors.white70),
+                              border: OutlineInputBorder(
+                                borderRadius: BorderRadius.all(Radius.circular(2)),
+                              ),
+                              enabledBorder: OutlineInputBorder(
+                                borderRadius: BorderRadius.all(Radius.circular(2)),
+                                borderSide: BorderSide(color: Colors.white24),
+                              ),
+                              focusedBorder: OutlineInputBorder(
+                                borderRadius: BorderRadius.all(Radius.circular(2)),
+                                borderSide: BorderSide(color: Colors.white),
+                              ),
+                            ),
+                            onChanged: _handleEntropyInput,
+                          ),
+                        ),
+                      ),
+                      const SizedBox(width: 8),
+                      Row(
+                        children: [
+                          SizedBox(
+                            height: 16,
+                            width: 16,
+                            child: Checkbox(
+                              value: _isHexMode,
+                              onChanged: (value) {
+                                setState(() {
+                                  _isHexMode = value ?? false;
+                                  if (_entropyController.text.isNotEmpty) {
+                                    _handleEntropyInput(_entropyController.text);
+                                  }
+                                });
+                              },
+                              side: const BorderSide(color: Colors.white70),
+                              fillColor: WidgetStateProperty.resolveWith<Color>(
+                                (Set<WidgetState> states) {
+                                  if (states.contains(WidgetState.selected)) {
+                                    return Colors.white24;
```

Comment:
Never use `Colors.` It will break in either light or dark mode.

If you want to hard-code a color use `final theme = SailTheme.of(context).colors.<color that makes sense here>`